src/veritysetup/veritysetup.c

```diff
@@ -109,11 +126,76 @@ static int parse_options(const char *options) {
+                else if ((val = startswith(word, "no-superblock="))) {
+
+                        r = parse_boolean(val);
+                        if (r < 0)
+                                return log_error_errno(r, "Failed to parse %s: %m", word);
+
+                        arg_no_superblock = r;
+                } else if ((val = startswith(word, "format="))) {
+
+                        r = safe_atoi(val, &arg_format);
+                        if (r < 0)
+                                return log_error_errno(r, "Failed to parse %s: %m", word);
+                } else if ((val = startswith(word, "data-block-size="))) {
+                        uint64_t sz;
+
+                        r = safe_atou64(val, &sz);
+                        if (r < 0)
+                                return log_error_errno(r, "Failed to parse %s: %m", word);
+                        if (!block_size_is_valid(sz))
+                                return log_error_errno(SYNTHETIC_ERRNO(EINVAL), "Failed to parse %s: %m", word);
+
+                        arg_data_block_size = sz;
+                } else if ((val = startswith(word, "hash-block-size="))) {
+                        uint64_t sz;
+
+                        r = parse_size(val, 1024, &sz);
+                        if (r < 0)
+                                return log_error_errno(r, "Failed to parse %s: %m", word);
+                        if (!block_size_is_valid(sz))
+                                return log_error_errno(SYNTHETIC_ERRNO(EINVAL), "Failed to parse %s: %m", word);
+
+                        arg_hash_block_size = sz;
+                } else if ((val = startswith(word, "data-blocks="))) {
+
+                        r = safe_atou64(val, &arg_data_blocks);
+                        if (r < 0)
+                                return log_error_errno(r, "Failed to parse %s: %m", word);
+                } else if ((val = startswith(word, "hash-offset="))) {
 
                         r = safe_atou64(val, &arg_hash_offset);
                         if (r < 0)
                                 return log_error_errno(r, "Failed to parse %s: %m", word);
+                } else if ((val = startswith(word, "salt="))) {
+
+                        if (isempty(val)) {
+                                arg_salt = mfree(arg_salt);
+                                arg_salt_size = 32;
+                        } else if (streq(val, "-")) {
+                                arg_salt = mfree(arg_salt);
+                                arg_salt_size = 0;
+                        } else {
+                                r = unhexmem(val, strlen(val), &arg_salt, &arg_salt_size);
+                                if (r < 0)
+                                        return log_error_errno(r, "Failed to parse %s: %m", word);
+                        }
+                } else if ((val = startswith(word, "uuid="))) {
+                        sd_id128_t id;
+
+                        r = sd_id128_from_string(val, &id);
```

Comment:
if you just want to validate the string and aren't interested in the actual value you can pass NULL as second arg here, in which case we'll parse as usual but not return what we parse but still return errros

Reply:
done.